src/hooks/UseForm/JobApplication/JobApplication.tsx

```diff
@@ -0,0 +1,145 @@
+import useForm, { Errors } from '../useForm'
+
+enum Title {
+  Mr = 'MR',
+  Mrs = 'MRS',
+  Miss = 'MISS',
+  Ms = 'MS',
+}
+
+export type ApplicationFormData = {
+  email: string
+  title: Title
+  name: string
+  startDate: Date
+  expectedSalary: number
+  hasAgreedToTerms: boolean
+}
+
+export type ApplicationErrors = Errors<ApplicationFormData>
+
+interface Props {
+  onSubmit: (data: ApplicationFormData) => void
+  onError: (data: ApplicationErrors) => void
+}
+
+const JobApplication: React.FC<Props> = ({ onSubmit, onError }) => {
+  const { register, errors, hasBegun, hasChangedWithoutSubmit, handleSubmit } =
+    useForm<ApplicationFormData>()
+
+  return (
+    <div className="flex w-96 flex-col gap-4">
+      <div className="flex flex-col gap-1">
+        <label className="block text-sm font-medium text-gray-900">
+          Your email
+          <input
+            type="email"
+            className="block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 "
+            placeholder="name@example.com"
+            {...register('email', {
+              isRequired: true,
+              validate: email => (email.includes('@') ? null : 'Email must include an @'),
+            })}
+          />
+        </label>
+        {errors?.email && <span className="text-red-800">{errors.email}</span>}
+      </div>
+      <div className="flex flex-col gap-1">
+        <Select
+          options={[
+            { label: 'Mr', value: Title.Mr },
+            { label: 'Mrs', value: Title.Mrs },
+            { label: 'Miss', value: Title.Miss },
+            { label: 'Ms', value: Title.Ms },
+          ]}
+          placeholder="Title"
+          {...register('title', { isRequired: true, refName: 'selectRef' })}
+        />
+        {errors?.title && <span className="text-red-800">{errors.title}</span>}
+      </div>
+      <div className="flex flex-col gap-1">
+        <label className="block text-sm font-medium text-gray-900">
+          Your name
+          <input
+            className="block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 "
+            {...register('name', { isRequired: true })}
+          />
+        </label>
+        {errors?.name && <span className="text-red-800">{errors.name}</span>}
+      </div>
+      <div className="flex flex-col gap-1">
+        <label className="block text-sm font-medium text-gray-900">
+          Available start date
+          <input
+            type="date"
+            className="block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 "
+            {...register('startDate', {
+              isRequired: true,
+              validate: startDate =>
+                startDate < new Date(Date.now()) ? 'Date must be in the future' : null,
```

Comment:
Yeah I agree we can add a lot more to our validators but I think it is fine to add those in later since it won't be a breaking change just added functionality. I'll open an issue once this goes in.